docs/tools/firmware.sh

```diff
@@ -43,72 +44,47 @@ case "$os" in
+		echo "Copying this script to the ESP"
+		cp "$0" "/Volumes/${EFILABEL}/firmware.sh"|| (echo -e "\nFailed to copy script.\nPlease copy the script manually to the EFI partition using Finder\n" && echo && read -p "Press enter after you have copied" && echo)
 		echo "Unmounting the EFI partition"
-		sudo diskutil unmount disk0s1
+		diskutil unmount "/Volumes/${EFILABEL}/"
 		echo
-		echo -e "Run the following commands or run this script itself in Linux now to set up Wi-Fi :-\n\nsudo umount /dev/nvme0n1p1\nsudo mkdir -p /tmp/apple-wifi-efi\nsudo mount /dev/nvme0n1p1 /tmp/apple-wifi-efi\nbash /tmp/apple-wifi-efi/firmware.sh\n"
+		echo -e "Run the following commands or run this script itself in Linux now to set up Wi-Fi:\n\n    mkdir -p /tmp/apple-wifi-efi\n    mount /dev/nvme0n1p1 /tmp/apple-wifi-efi\n    /tmp/apple-wifi-efi/firmware.sh\n"
 		;;
 	(Linux)
 		echo "Detected Linux"
 		echo "Re-mounting the EFI partition"
+		mountpoint=$(mktemp -d)
+		workdir=$(mktemp -d)
+		echo "Installing Wi-Fi and Bluetooth firmware"
 		if [[ ${1-default} = -v ]]
 		then
-			sudo umount -v /dev/nvme0n1p1 || true
-			sudo mkdir -p -v /tmp/apple-wifi-efi || true
-			sudo mount -v /dev/nvme0n1p1 /tmp/apple-wifi-efi || true
+			mount -v /dev/nvme0n1p1 $mountpoint
+			tar --warning=no-unknown-keyword -xvC $workdir -f $mountpoint/firmware.tar.gz
+    	python3 $0 $workdir $workdir/firmware-renamed.tar
+			tar -xvC /lib/firmware -f $workdir/firmware-renamed.tar
 		else
-			sudo umount /dev/nvme0n1p1 2>/dev/null || true
-			sudo mkdir -p /tmp/apple-wifi-efi 2>/dev/null || true
-			sudo mount /dev/nvme0n1p1 /tmp/apple-wifi-efi 2>/dev/null || true
+			mount /dev/nvme0n1p1 $mountpoint
+			tar --warning=no-unknown-keyword -xC $workdir -f $mountpoint/firmware.tar.gz
+    	python3 $0 $workdir $workdir/firmware-renamed.tar &> /dev/null
+			tar -xC /lib/firmware -f $workdir/firmware-renamed.tar
 		fi
-		mountpoint=$(findmnt -n -o TARGET /dev/nvme0n1p1)
-		echo "Getting Wi-Fi and Bluetooth firmware"
-		sudo mkdir -p /tmp/apple-wifi-fw
-		cd /tmp/apple-wifi-fw
-		if [[ ${1-default} = -v ]]
-		then
-			sudo tar xvf $mountpoint/firmware.tar.gz
-		else
-			sudo tar xf $mountpoint/firmware.tar.gz 2>/dev/null
-		fi
-		echo "Setting up Wi-Fi and Bluetooth"
-		if [[ ${1-default} = -v ]]
-		then
-			sudo python3 $mountpoint/firmware.sh /tmp/apple-wifi-fw
-		else
-			sudo python3 $mountpoint/firmware.sh /tmp/apple-wifi-fw >/dev/null
-		fi
-		cd /lib/firmware
-		if [[ ${1-default} = -v ]]
-		then
-			sudo tar xvf /tmp/apple-wifi-fw/firmware.tar
-		else
-			sudo tar xf /tmp/apple-wifi-fw/firmware.tar
-		fi
-
+		echo "Installing extra firmware"
 		for file in "$mountpoint/brcmfmac4364b2-pcie.txt" \
-		            "$mountpoint/brcmfmac4364b2-pcie.txcap_blob"
+			"$mountpoint/brcmfmac4364b2-pcie.txcap_blob"
 		do
 			if [ -f "$file" ]
 			then
 				if [[ ${1-default} = -v ]]
 				then
-					sudo cp -v $file /lib/firmware/brcm
```

Comment:
Why have you remove sudo from various places like this

Reply:
I changed the instructions to run the whole script with sudo, since some sudo configs have a really short timeout.

Reply:
Reverted